api/osb/osb_store_plugin.go

```diff
@@ -0,0 +1,1113 @@
+/*
+* Copyright 2018 The Service Manager Authors
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+
+package osb
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/tidwall/sjson"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/Peripli/service-manager/pkg/query"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/Peripli/service-manager/pkg/util"
+
+	"github.com/Peripli/service-manager/storage"
+
+	"github.com/Peripli/service-manager/pkg/log"
+
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+const (
+	// OSBStorePluginName is the plugin name
+	OSBStorePluginName = "OSBStorePluginName"
+	smServicePlanIDKey = "sm_service_plan_id"
+	smContextKey       = "sm_context_key"
+)
+
+type entityOperation string
+
+const (
+	READY    entityOperation = "ready"
+	ROLLBACK entityOperation = "rollback"
+	DELETE   entityOperation = "failed"
+	NONE     entityOperation = "none"
+)
+
+type BindingStorageKey struct{}
```

Comment:
To ensure encapsulation context keys should use a private type and provide strongly-typed access functions to specific values. See pkg/web/context_keys.go

Reply:
Also add tests for this context value
You can do that by registering a filter that sets this context value depending on some criteria, e.g. global var. You can register this filter in test/osb_test/osb_suite_test.go in `WithSMExtensions`.

Reply:
done